meeshkan/serve/mock/server.py

```diff
@@ -25,32 +28,57 @@ def make_mocking_app_(
     return Application([(r"/.*", MockServerView, dependencies)])
 
 
-def make_mocking_app(callback_dir: Optional[str], specs_dir: str, routing: Routing):
-
+def make_mocking_app(
+    callback_dir: Optional[str], specs: List[OpenAPISpecification], routing: Routing
+):
     # callback_manager = CallbackManager()
     if callback_dir is not None:
         callback_manager.load(callback_dir)
 
-    response_matcher = ResponseMatcher(specs_dir)
+    response_matcher = ResponseMatcher(specs)
 
     return make_mocking_app_(callback_manager, response_matcher, routing)
 
 
 class MockServer:
     def __init__(
-        self, port, specs_dir, callback_dir=None, admin_port=None, routing=PathRouting()
+        self, port, specs, callback_dir=None, admin_port=None, routing=PathRouting()
     ):
         self._callback_dir = callback_dir
         self._admin_port = admin_port
         self._port = port
-        self._specs_dir = specs_dir
+        self._specs = specs
         self._routing = routing
 
-    def run(self):
+    def run(self) -> None:
         if self._admin_port:
             start_admin(self._admin_port)
-        app = make_mocking_app(self._callback_dir, self._specs_dir, self._routing)
+        app = make_mocking_app(self._callback_dir, self._specs, self._routing)
         http_server = HTTPServer(app)
         http_server.listen(self._port)
-        logger.info("Mock is listening on http://localhost:%s", self._port)
+        self.log_startup()
         IOLoop.current().start()
+
+    def log_startup(self) -> None:
+        for spec in self._specs:
+            if not spec.api.servers:
+                continue
+            for path, path_item in spec.api.paths.items():
+                for method in [
+                    "get",
+                    "put",
+                    "post",
+                    "delete",
+                    "options",
+                    "head",
+                    "patch",
+                    "trace",
+                ]:
+                    if getattr(path_item, method):
+                        logger.info(
+                            "→ "
+                            + method.upper().ljust(7)
+                            + f" http://localhost:{self._port}/"
+                            + spec.api.servers[0].url
+                            + path
```

Comment:
Could this be huge for servers like Stripe?

Reply:
The URL and path? I guess it can be long, but I think we should still log it (even though a bit ugly, a long and ugly URL is even more motivated to log).

Reply:
I mean there could be hundreds of operations supported by the path? It's a very good idea to log them, just wondering if the initialization messages should then appear after them...

Reply:
Aha, if the number of endpoints is big. Hm, not sure, perhaps we can still log them all (so user can scroll or grep through them) until we have some other idea or suggestion?